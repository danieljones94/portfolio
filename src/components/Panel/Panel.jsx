import React, { Component } from "react";
import styles from "./Panel.module.scss";

class Panel extends Component {
  state = {};
  render() {
    return (
      <article className={styles.panelContainer}>
        <p>{this.props.title}</p>
        <p>{this.props.description}</p>
      </article>
    );
  }
}

export default Panel;
