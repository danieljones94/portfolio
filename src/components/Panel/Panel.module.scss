.panelContainer {
  width: 40%;
  margin-top: 5vh;
  box-shadow: 0 15px 10px rgb(35, 54, 66);
  border: none;
  border-radius: 5%;

  .panel {
    height: 100%;
    width: 100%;
    border: none;
    border-radius: 5%;
  }
}
