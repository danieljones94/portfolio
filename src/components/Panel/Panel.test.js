import React from "react";
import Panel from "./Panel";
import { mount } from "enzyme";

const mockURL = "http://nug-of-sauce.danmjones.co.uk/";
const mockDescription =
  "A tug of war game, whereby the player who presses their button the most times wins.";

describe("Panel tests", () => {
  let testFunction;
  let component;

  beforeEach(() => {
    testFunction = jest.fn();
    component = mount(
      <Panel
        url={mockURL}
        onMouseEnter={() => testFunction()}
        description={mockDescription}
      />
    );
  });

  it("should match the snapshot", () => {
    expect(component).toMatchSnapshot();
  });

  it("should contain a HTML p tag", () => {
    expect(component.find("p").length).toEqual(2);
    expect(component.find("a").length).toEqual(1);
    expect(component.find("img").length).toEqual(1);
  });

  it("should receive data that is passed down as props", () => {
    expect(component.props().url).toEqual(mockURL);
  });

  it("should render data passed down as props", () => {
    expect(component.text()).toContain(mockDescription);
  });

  it("should toggle the hover state when the enters the component", () => {
    expect(component.state("hover")).toEqual(false);
    component.instance().toggleHover();
    expect(component.state("hover")).toEqual(true);
  });
});
