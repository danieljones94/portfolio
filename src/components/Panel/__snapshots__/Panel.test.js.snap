// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel tests should match the snapshot 1`] = `
<Panel
  description="A tug of war game, whereby the player who presses their button the most times wins."
  onMouseEnter={[Function]}
  url="http://nug-of-sauce.danmjones.co.uk/"
>
  <section
    className="panelContainer"
  >
    <a
      href="http://nug-of-sauce.danmjones.co.uk/"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        className="panel"
        style={
          Object {
            "color": "transparent",
            "cursor": "pointer",
            "padding": "3vh",
          }
        }
      />
      <p
        style={
          Object {
            "color": "transparent",
            "cursor": "pointer",
            "padding": "3vh",
          }
        }
      >
        A tug of war game, whereby the player who presses their button the most times wins.
      </p>
      <p
        style={
          Object {
            "color": "transparent",
            "cursor": "pointer",
            "padding": "3vh",
          }
        }
      />
    </a>
  </section>
</Panel>
`;
